### PR TITLE
Docs/api references extended

### PR DESCRIPTION
**Pull request by bartvanb**
This adds the API references for cli/common/client and adds extra to existing node/client API refs

### CODE REVIEW

File: vantage6-common/vantage6/common/configuration_manager.py [hunk 1]
```diff
@@ -79,56 +143,175 @@ def put(self, env: str, config: dict):
...
             config = yaml.safe_load(f)
 
         for env in self.ENVS:
             self.put(env, self._get_environment_from_dict(config, env))
 
     @classmethod
-    def from_file(cls, path, conf_class=Configuration):
+    def from_file(
+        cls, path: Path | str, conf_class: Type[Configuration] = Configuration
+    ) -> 'ConfigurationManager':
```
Comment (frankcorneliusmartin) [inline vantage6-common/vantage6/common/configuration_manager.py:277]: See comment about classmethod return types

File: vantage6-common/vantage6/common/context.py [hunk 2]
```diff
@@ -205,25 +316,73 @@ def log_file(self):
...
+        Returns
+        -------
+        str
+            Environment
+        """
         return self.__environment
 
     @environment.setter
-    def environment(self, env):
+    def environment(self, env) -> None:
```
Comment (frankcorneliusmartin) [inline vantage6-common/vantage6/common/context.py:371]: Type `str`missing

**Review by frankcorneliusmartin — CHANGES_REQUESTED** [review]
Lot of work Bart! Let merge this into 3.9 as we want to test is sufficiently

**Review by frankcorneliusmartin — CHANGES_REQUESTED** [review]
Lot of work Bart! Let merge this into 3.9 as we want to test is sufficiently

File: tools/update-version.py [hunk 3]
```diff
@@ -3,65 +3,139 @@
...
+    """
     print(msg)
 
-pattern = r"(version_info\s*=\s*\(*)(\s*\d+,\s*\d+,\s*\d+,)(\s*)('\w*')(,\s*__build__)(,\s*)(\d+)(\)*)"
+
+# pattern to use in the regex to update version
+pattern = (
+    r"(version_info\s*=\s*\()(\s*\d+,\s*\d+,\s*\d+,)(\s*)('\w*')"
+    r"(,\s*__build__)(,\s*)(\d+)(\))"
+)
```
[the remark this answers is not in the document]
Comment (bartvanb) [reply tools/update-version.py:23]: This is actually a change in release/3.9. So not sure how that ended up here.

File: vantage6-client/vantage6/client/algorithm_client.py [hunk 4]
```diff
@@ -160,8 +168,8 @@ def create(
             ----------
             input_ : bytes
                 Input to the task. Should be b64 encoded.
-            organization_ids : List[int], optional
-                List of organization IDs that should execute the task
+            organization_ids : list[int]
+                List of organization IDs that should execute the task.
```
[the remark this answers is not in the document]
Comment (bartvanb) [reply vantage6-client/vantage6/client/algorithm_client.py]: Well it still has a default value.. but specifying zero organizations doesn't make sense. A task for 0 orgs will never be executed by anyone so why create it?

File: vantage6-client/vantage6/tools/dispatch_rpc.py [hunk 5]
```diff
@@ -2,13 +2,41 @@
...
+from types import ModuleType
+from typing import Any
+
 from vantage6.client import ContainerClient
 from vantage6.client.algorithm_client import AlgorithmClient
 from vantage6.tools.util import info, warn, error
 
 
-def dispatch_rpc(data, input_data, module, token, use_new_client=False):
+def dispatch_rpc(data: Any, input_data: dict, module: ModuleType, token: str,
```
[the remark this answers is not in the document]
Comment (bartvanb) [reply vantage6-client/vantage6/tools/dispatch_rpc.py]: You're right, it should be str. It is used to obtain a ModuleType and I guess I misread

File: vantage6-server/vantage6/server/cli/server.py [hunk 6]
```diff
@@ -246,9 +307,21 @@ def cli_server_import(ctx, file_, drop_all):
...
 @cli_server.command(name='shell')
 @click_insert_context
-def cli_server_shell(ctx):
-    """ Run a iPython shell. """
-    # shell.init(ctx.environment)
+def cli_server_shell(ctx: ServerContext) -> None:
+    """
+    Run an iPython shell.
+
+    Note that using the shell is not recommended, as there are no checks on
```
[the remark this answers is not in the document]
Comment (bartvanb) [reply vantage6-server/vantage6/server/cli/server.py]: Nice addition, done!

File: vantage6-server/vantage6/server/context.py [hunk 7]
```diff
@@ -1,29 +1,46 @@
...
+    Note that this context is specific to the server: for nodes, there is a
+    separate test context.
+    """
 
     INST_CONFIG_MANAGER = TestingConfigurationManager
     LOGGING_ENABLED = False
 
     @classmethod
-    def from_external_config_file(cls, path):
+    def from_external_config_file(cls) -> 'TestContext':
```
[the remark this answers is not in the document]
Comment (bartvanb) [reply vantage6-server/vantage6/server/context.py]: this has been deleted there in 3.9

**Pull request merged**